Remove more hardcoded `master` references

jeffwidman opened the pull request:
Follows up on https://github.com/dependabot/dependabot-core/pull/6149 by removing two more hardcoded references to `master` that I discovered.

Originally I assumed we'd need to flip these to `HEAD`, but in https://github.com/dependabot/dependabot-core/pull/6149 we were able to get away with simply removing the hardcoded reference altogether in the `bundler` ecosystem and we haven't seen any ill effects the past 6 months...

So let's try removing this and see if it works as smoothly in these ecosystems as it did in the `bundler` ecosystem.

Fix https://github.com/dependabot/dependabot-core/issues/6202

jeffwidman commented:
And two code failures... since these are two separate ecosystems, I'll likely break this into two separate PR's...

I suspect this indicates we'll need to still hardcode some value, but perhaps we can get away with `HEAD`, otherwise we'll have to have the `master` vs `main` conversation... 😩

jeffwidman commented on hex/helpers/lib/parse_deps.exs:
```diff
@@ -109,7 +109,7 @@ defmodule Parser do
     %{
       type: "git",
       url: repo_url,
-      branch: opts[:branch] || "master",
```
@TylerWitt any idea why this might be failing? You seem to be a lot more knowledgable than me about Hex/Erlang... 

If we can figure out a fix, should probably split this out to it's own dedicated PR...

TylerWitt replied:
Sorry for the delay!

This might have been a temporary hex outage or other transient error. I cherry-picked this commit and the build worked fine. I'd retry CI.

jeffwidman replied:
No worries, thanks for trying--your timing is impeccable... you probably tried it just a few days after it was fixed 😁 :
* https://github.com/dependabot/dependabot-core/pull/12109

jeffwidman commented:
Now that the Hex fix was handled in a separate PR, I've rebased this and updated the PR / commit description to reflect that the scope of this is now only for `npm`.

kbukum1 commented:
@jeffwidman , 

Do we want to merge this PR? If we want, I can approve it and deploy it.

kbukum1 reviewed (approved):
LGTM
